credential/src/lib.rs

```diff
@@ -123,6 +137,12 @@ pub trait ProvideAwsCredentials {
     fn credentials(&self) -> Result<AwsCredentials, CredentialsError>;
 }
 
+impl ProvideAwsCredentials for AwsCredentials {
```

Comment:
it was used in tests/sts.rs but only in passing and i'll remove that.
it can be used as a way of injecting basic credentials into a client without a further wrapper type that itself implements ProvideAwsCredentials.